### PR TITLE
Feature/add lookback macro

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
**What change does this PR introduce?** 
 

> - Added the new `fivetran_lookback()` macro. This generates SQL to simplify adding lookback windows to incremental models. See [README](https://github.com/fivetran/dbt_fivetran_utils/blob/releases/v0.4.latest/README.md#fivetran_lookback-source).
> - Added the new `is_databricks_sql_warehouse()` macro. This determines if Databricks destination is a SQL Warehouse or All Purpose Cluster. See [README](https://github.com/fivetran/dbt_fivetran_utils/blob/releases/v0.4.latest/README.md#is-databricks-sql-warehouse-source).

**If this PR introduces a new macro, how did you test the new macro?** 

- Tested in my updates for Jira, Netsuite, and Shopify. I also made sure the check the generated compiled code for each warehouse to ensure it was generating the SQL as expected. 

**If this PR introduces a modification to an existing macro, which packages is the macro currently present in and what steps were taken to test compatibility across packages?** 

N/A

**Did you update the README to reflect the macro addition/modifications?** 

- [x] Yes
- [ ] No (provide further explanation)

### CODE REVIEW

File: integration_tests/packages.yml
Comment (fivetran-joemarkiewicz) [inline integration_tests/packages.yml:8]: Would it be possible to swap this before release?

File: macros/fivetran_lookback.sql
Comment (fivetran-joemarkiewicz) [inline macros/fivetran_lookback.sql:14]: I actually don't have a recollection of using the `dbt_utils.get_single_value()` macro before. Is this essentially doing the same thing as this [run_query](https://github.com/fivetran/dbt_salesforce/blob/fa353308688ed708ea2da5f8116f21ec8aba4989/models/salesforce/intermediate/int_salesforce__date_spine.sql#L13) operation? If so, the get_sing_value is much cleaner and probably what we should be using across our models!

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-catfritz this is looking great and I don't have any comments on your code changes that would require changes from your end before approval. The only thing I would request is for you to run buildkite and ensure all tests pass. Once they do, I can come in and give this an approval!

File: macros/fivetran_lookback.sql
Comment (fivetran-catfritz) [reply macros/fivetran_lookback.sql:14]: It is slightly different execution, but the end result looks like it's practically the same. [Here](https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/get_single_value.sql) is the source for `get_single_value()`. I found it by accident while reading documentation and thought I'd give it a try to make things easier.

File: integration_tests/packages.yml
Comment (fivetran-catfritz) [reply integration_tests/packages.yml:8]: Only if `dbt_shopify_holistic_reporting` is updated with the newest shopify package versions.

File: macros/is_databricks_sql_warehouse.sql
Comment (fivetran-jamie) [inline macros/is_databricks_sql_warehouse.sql]: since this is currently in use in the [Fivetran Platform package](https://github.com/fivetran/dbt_fivetran_log/blob/main/macros/is_databricks_sql_warehouse.sql), should we adjust the name to `fivetran_is_databricks_sql_warehouse` to not cause any conflicts? and then we can update the reference in Fivetran Platform to use this version of the macro (and delete the old one)

otherwise i think we'll need to add an adapter.dispatch line

@fivetran-catfritz

File: README.md
Comment (fivetran-jamie) [inline README.md]: is this actually necessary to have as an argument? like if it always must be `target` it might be harmful to have this as a parameter? cc @fivetran-joemarkiewicz as the macro author

File: README.md
Comment (fivetran-jamie) [inline README.md]:
```suggestion
    {{ fivetran_utils.fivetran_lookback(
```

File: README.md
Comment (fivetran-jamie) [inline README.md]: @fivetran-catfritz could you switch the order of `interval` and `datepart` to match the macro? the order of arguments in the macro file is `from_date, datepart, interval, safety_date`

File: macros/is_databricks_sql_warehouse.sql
Comment (fivetran-joemarkiewicz) [reply macros/is_databricks_sql_warehouse.sql]: YES! Let's rename to be safe and consistent

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md]: That's a good point. Especially since the target is always accessible regardless if it is a parameter or not. We can probably remove this as a parameter.

Comment (fivetran-catfritz) [reply README.md]: I tried two versions of the macro in my local Shopify to remove the argument and neither would compile with the following errors:
1. `{% macro is_databricks_sql_warehouse(target=target) %}`
    - Error: `Compilation Error` ... `'target' is undefined`
2. `{% macro is_databricks_sql_warehouse() %}`
    - Error: `Compilation Error`... `macro 'dbt_macro__is_databricks_sql_warehouse' takes not more than 0 argument(s)`

I wasn't surprised since I have had this issue in the past trying to create arg-less macros, and searching around I haven't found a solution. @fivetran-joemarkiewicz might this be why you opted in include the target object as an argument? @fivetran-jamie Do you have any ideas on how to get this to work?

Comment (fivetran-jamie) [reply README.md]: hm how did you test these? for the second one, did you remove the `target` argument from where you call the macro? the error message kinda sounds like an extra/un-needed argument has been provided to a macro call

File: README.md
Comment (fivetran-catfritz) [inline README.md:501]:
```suggestion
* `datepart` (required): The grain of the interval. 
* `interval` (required): The number of units to look backwards.
```

File: README.md
Comment (fivetran-catfritz) [inline README.md]:
```suggestion
* `datepart` (required): The grain of the interval. 
* `interval` (required): The number of units to look backwards.
* `safety_date` (optional): This date will be used in the rare case that the `from_date` expression returns a null value. This only needs to be specified if you want to change the default value from '2010-01-01'.
```

File: macros/is_databricks_sql_warehouse.sql
Comment (fivetran-catfritz) [inline macros/is_databricks_sql_warehouse.sql]:
```suggestion
{% macro fivetran_is_databricks_sql_warehouse(target) %}
```

File: README.md
Comment (fivetran-catfritz) [reply README.md]: I got it working! Updated!

File: README.md
Comment (fivetran-catfritz) [reply README.md]: Updated.

File: macros/is_databricks_sql_warehouse.sql
Comment (fivetran-catfritz) [reply macros/is_databricks_sql_warehouse.sql]: Updated.

**Comment by fivetran-catfritz** [comment]
Thanks for your suggestions @fivetran-jamie ! I have updated.

**Review by fivetran-jamie — APPROVED** [review]

**Pull request merged**
